src/vm/diagnosticserver.h

```diff
@@ -16,6 +16,7 @@ enum class DiagnosticMessageType : uint32_t
     ///////////////////////////////////////////////////////////////////////////
     // Debug = 0
     GenerateCoreDump = 1,           // Initiates core dump generation
+    AttachProfiler,
```

Comment:
My implementation code (see #24582) uses a different set of codes than this, but it should be as trivial as the modifications I made to Mike's GenerateCoreDump logic since it is nearly identical.  If this gets merged in before that, I can modify this to bring it in line and document it in the spec.

Reply:
I just missed the comment about reserved numbers for the profiler. I'll change it for now to respect that and then we can make it work with John's changes